### ISSUE TITLE
Using macros with :npm-module doesn't pull the required modules into the output .js

### ISSUE THREAD

**Issue by caleb** [thheller/shadow-cljs#426]
Hi,

I am having issues with using macros with my `:npm-module`. If I use a macro, the macro code is expanded in the output javascript file, but if that expanded code uses other modules that my original module didn't include, then I get an error when I run the resulting javascript (e.g. `Can't find variable: sablono`)

Let's say I have a module with a macro:

```clojure
;; utils.clj
(ns utils
  (:require [sablono.core :as sablono])

(defmacro my-macro []
  `(sablono/html [:div "Look at my div"]))
```

And then I want to use that macro from my Clojurescript file:

```clojure
;; core.cljs
(ns core
  (:require-macros [utils :as utils]))

(utils/my-macro)
```

This results in an error because the macro is expanded and the code `(sablono/html [:div "Look at my div"])` gets put in my `core` module, but in the output javascript the `sablono` dependency isn't required so I get the error `Can't find variable: sablono` when I run it.

If I include the `sablono` dependency myself in `core`, then it works:

```clojure
;; core.cljs
(ns core
  (:require-macros [utils :as utils])
  (:require [sablono.core :as sablono]))

(utils/my-macro)
```

I've tested this with the `:browser` output and it works fine.

What do you think?

(PS thanks for shadow-cljs, it's rejuvenated my interest in Clojurescript since it is so easy to use)

**Comment by thheller** [comment]
`:browser` works since it only has one global scope. `:npm-module` has its own scope per file and only imported what was explicitly required. I changed it so that all potential namespace should be available properly.

Try `2.7.12`.

**Comment by caleb** [comment]
`2.7.12` seems to work on that front, but now I'm getting another javascript error:

<img width="1132" alt="screen shot 2019-01-06 at 1 05 20 pm" src="https://user-images.githubusercontent.com/4692/50739860-d1e86f80-11b3-11e9-8aae-e3ba6456d15d.png">

It doesn't seem to like the dash in my module name.

Here's my whole Clojurescript program:

```clojure
;; src/cljs/main/mojo_tools/frontend/main.cljs
(ns mojo-tools.frontend.main
  (:require [rum.core :as rum]
            ["@atlaskit/datetime-picker" :as datetime-picker]))

(defonce state (atom "My State"))

(rum/defc label [text]
  [:div {:class "label"} text])

(defn start []
  (js/console.log "Started"))

(defn stop [done]
  (js/console.log "Stopped")
  (done))

(rum/mount (label (str "State:" @state))
             (js/document.getElementById "expensable-form"))
```

**Comment by thheller** [comment]
Yeah, I didn't munge the names properly so it didn't turn the `-` into `_`. Should be fixed in `2.7.13`.
